m4/angry_bird.py

```
@@ -0,0 +1,241 @@
+        for block in self.blocks:
+            if not block.destroyed:
+                # Bug 4: 简化的碰撞检测错误
+                if (bird.x > block.x and bird.x < block.x + block.width and
+                    bird.y > block.y and bird.y < block.y + block.height):
+                    block.destroyed = True
+                    bird.vx *= -0.5  # 反弹
+                    bird.vy *= -0.5
+        
+        # Bug 5: 地面碰撞检测错误
+        if bird.y > SCREEN_HEIGHT - GROUND_HEIGHT:
+            bird.y = SCREEN_HEIGHT - GROUND_HEIGHT  # 错误：没有考虑小鸟半径
```

Ruff (0.13.1)

Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)